library/network_connections.py

```diff
@@ -1789,6 +1861,14 @@ def run_prepare(self):
             connection["nm.exists"] = exists
             connection["nm.uuid"] = uuid
 
+    def finish_run_error(self, idx, action, error):
+        Cmd.finish_run_error(self, idx, action, error)
+        self.nmutil.rollback_checkpoint(self._checkpoint)
+
+    def finish_run_success(self):
```

Comment:
I would call these `transaction_commit()` and `transaction_rollback()`. Of course, it's not a real "transaction" as we know it from data bases (ACID), but I think the term is better. At least, we also need a "start()" method, and the name of the start method should be reminiscent of the accompanying finish-run methods.

Reply:
renamed/added start/rollback/finish methods for a transaction.